src/test/java/HandleGetRequestTest.java

```diff
@@ -2,9 +2,9 @@
 import java.util.*;
 import java.io.IOException;
 
-public class HttpHandlerSelectorTest extends junit.framework.TestCase {
+public class HandleGetRequestTest extends junit.framework.TestCase {
 
-  HttpHandlerSelector handler;
+  HandleGetRequest getRequest;
```

Comment:
This is confusing. There is a `getRequest`, but there is also a `request` which appears to always be a GET request ... you see what I'm saying?